src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -112,29 +165,70 @@ public static object ConvertResult(JsonOperationContext ctx, object result)
 
-        public override unsafe (long Index, object Value) Execute(TransactionOperationContext context, Table items, long index)
+        public RemoveCompareExchangeCommand(string database, string key, long index, JsonOperationContext contextToReturnResult, bool fromBackup) : base(database, key,
+            index, contextToReturnResult)
+        {
+            _fromBackup = fromBackup;
+        }
+
+        public override unsafe (long Index, object Value) Execute(TransactionOperationContext context, Table items, long index, Table tombstoneItems, bool fromBackup = false)
         {
             using (Slice.From(context.Allocator, ActualKey, out Slice keySlice))
             {
                 if (items.ReadByKey(keySlice, out var reader))
                 {
+                    if (fromBackup)
+                    {
+                        items.Delete(reader.Id);
+                        return (index, null);
+                    }
                     var itemIndex = *(long*)reader.Read((int)ClusterStateMachine.UniqueItems.Index, out var _);
                     var storeValue = reader.Read((int)ClusterStateMachine.UniqueItems.Value, out var size);
                     var result = new BlittableJsonReaderObject(storeValue, size, context);
+
                     if (Index == itemIndex)
                     {
                         result = result.Clone(context);
                         items.Delete(reader.Id);
+                        WriteCompareExchangeTombstone(context, tombstoneItems, index);
                         return (index, result);
                     }
                     return (itemIndex, result);
                 }
             }
             return (index, null);
         }
+
+        private unsafe void WriteCompareExchangeTombstone(TransactionOperationContext context, Table tombstoneItems, long index)
+        {
+            GetKeyAndPrefixIndexSlices(context, Database, Key, index, out var keyTuple, out var indexTuple);
+
+            using (keyTuple.Scope)
+            using (indexTuple.Scope)
+            using (Slice.External(context.Allocator, keyTuple.Buffer.Ptr, keyTuple.Buffer.Length, out var keySlice))
+            using (Slice.External(context.Allocator, indexTuple.Buffer.Ptr, indexTuple.Buffer.Length, out var indexSlice))
+            using (tombstoneItems.Allocate(out var tvb))
+            {
+                tvb.Add(keySlice.Content.Ptr, keySlice.Size);
+                tvb.Add(indexSlice);
+                tombstoneItems.Set(tvb);
+            }
+        }
+
+        public override DynamicJsonValue ToJson(JsonOperationContext context)
+        {
+            var json = base.ToJson(context);
+            json[nameof(Key)] = Key;
+            json[nameof(Index)] = Index;
+            json[nameof(Database)] = Database;
+            json["FromBackup"] = _fromBackup;
```

Comment:
Need to create a property with this name for proper deserialization?

Reply:
done